tools/common/schema/handler.go

```diff
@@ -92,8 +96,20 @@ func validateSetupConfig(config *SetupConfig) error {
 }
 
 func validateUpdateConfig(config *UpdateConfig) error {
-	if len(config.SchemaDir) == 0 {
-		return NewConfigError("missing " + flag(CLIOptSchemaDir) + " argument ")
+	if len(config.SchemaDir) == 0 && len(config.SchemaName) == 0 {
+		return NewConfigError("missing argument; either" + flag(CLIOptSchemaDir) + " or " +
+			flag(CLIOptSchemaName) + " must be specified")
+	}
+	if len(config.SchemaDir) > 0 && len(config.SchemaName) > 0 {
+		return NewConfigError("either" + flag(CLIOptSchemaDir) + " or " +
+			flag(CLIOptSchemaName) + " must be specified")
+	}
+	if len(config.SchemaName) > 0 {
+		if !slices.Contains(ValidSchemaNames["sql"], config.SchemaName) &&
+			!slices.Contains(ValidSchemaNames["cassandra"], config.SchemaName) {
+			return NewConfigError(flag(CLIOptSchemaDir) + " must be one of: " +
+				fmt.Sprintf("%v or %v", ValidSchemaNames["sql"], ValidSchemaNames["cassandra"]))
```

Comment:
Can you do this separately based on whether they'd chosen SQL or Cassandra? That way they're only given the schemas relevant for the tool at hand. Otherwise our `sql` tool print out schemas that should only be available for cassandra (and vice-versa)

Reply:
i could i just didnt want to "overengineer"..

Reply:
thinking about this more, I feel like the added messiness in our code to pass through the db type (we would have to kind of break the "common" db schema abstraction) is not worth the benefit to the user, since I think they will be able to figure out from the error message that the "cassandra" options aren't valid for their config if they're using sql, or the other way around.

Reply:
Our self-hosting users use this tools and I think its worth it to remove a potential source of confusion, but I wont hold this PR up over it.

I think this can be done simply by adding `Name() string` function to the `DB` interface used by this handler code. If you do that we could ask the provided DB object what it is when checking our configuration and only return the schema names that matter.

It'd be maybe 10 lines of code overall:
- The additional `Name` method definition
- ~3 lines in each of `tools/cassandra/cqlclient.go` and `tools/sql/conn.go` to add the `Name` implementations
- A line or two to get the DB's name and check it when validating configuration